evol/population.py

```diff
@@ -338,6 +339,42 @@ def callback(self, callback_function: Callable[..., None],
         callback_function(copy(self), **kwargs)
         return self
 
+    def group(self, grouping_function: Callable[..., List[List[int]]] = group_random, **kwargs) -> List['Population']:
```

Comment:
I'm not sure if I like this. It feels like a break in the pattern to have a verb not return a population but a list of populations instead. Would it not be more elegant, API wise, to have `pop.group()` just attach information to the resulting population that tells a `repeat_grouped`-type verb how to deal with the islands?

Reply:
I would like to keep all functionality of the population available also when not using an evolution object. So if we store the information on the islands internally then we need to alter the functionality of most methods of the population such that they can handle islands --  which would probably increase the complexity a lot.

I do not mind the difference in signature so much. After all, that is what it does: it converts a population into a few populations.

Reply:
@koaning , after our offline discussion, do you agree that this is the simplest/best solution?